fee_allocator/accounting/distribution.py

```diff
@@ -117,60 +175,18 @@ def re_distribute_incentives(
-    # As a result we have generated some debt that we owe the aura market in order to maintain the AURA/BAL split
-    # So now we find pools that have room to reallocate and adjust their split to restore the balance
-    if debt_to_aura_market:
-        debt_repaid = 0
-        ## Find how many pools ever could be over min_aura_incentive
-        pools_over_aura_min = [
-            pool_id
-            for pool_id, _data in incentives.items()
-            if _data["aura_incentives"] >= min_aura_incentive
-        ]
-        num_pools_over_min = len(pools_over_aura_min)
-        ## Figure out how much to shift per pool using an even split
-        if num_pools_over_min == 0:
-            print(
-                f"WARNING: {incentives[pool_id]['chain']}:{pool_id} has no pools over min_aura_incentive, but owes {debt_to_aura_market} to the aura market.  Debt will not be repaid."
-            )
-            amount_per_pool = 0
-        else:
-            amount_per_pool = round(debt_to_aura_market / num_pools_over_min, 4)
-        for pool_id in pools_over_aura_min:
-            ## TODO: Consider this logic as an additional test/more sensitive handlingthat could allow pool selection based
-            #   on total_incentives instead of aura incentives
-            #   if (incentives['aura_incentives'] + amount_per_pool) < min_aura_incentive:
-            #         num_pools_over_min -= 1
-            # Distribute the aura_debt to the pools that are over the min_aura_incentive
-            if incentives[pool_id]["total_incentives"] > 0:
-                # TODO:  Need to think about edge cases here and watch them.
-                incentives[pool_id]["aura_incentives"] += min(
-                    amount_per_pool, incentives[pool_id]["bal_incentives"]
-                )
-                incentives[pool_id]["bal_incentives"] -= min(
-                    amount_per_pool, incentives[pool_id]["bal_incentives"]
-                )
-                debt_repaid += min(
-                    amount_per_pool, incentives[pool_id]["bal_incentives"]
-                )
-            if debt_to_aura_market - debt_repaid >= 0:
-                print(
-                    f"{incentives[pool_id]['chain']}:{pool_id}  remaining debt to aura market: {debt_to_aura_market}, Debt repaid: {debt_repaid}, debt remaining: {debt_to_aura_market - debt_repaid}"
-                )
-    return incentives
+    # Now after everything is done, we need to make sure that all pools have at least min_aura_incentive
+    # if not we need to redistribute all aura_incentives to bal_incentives for that pool and keep track of how much has been reallocated
+
+    first_pass_buffer = 0.25
+    # First redistribute with a % buffer.
+    print(f"Redistributing Aura with a {first_pass_buffer} buffer.")
+    result = handle_aura_min(
+        incentives, min_aura_incentive * Decimal(1 - first_pass_buffer)
+    )
+    # Now redistribute again with no buffer
```

Comment:
nice solution!

i guess an even more inclusive approach would be to do this incrementally, so do a pass at 10%, 20%, 30% ... all the way up to 100%?

Reply:
Yeah we could, and maybe shoulkd later.  I don't think it will make any diference right now tho, In general there's not that many pools left.

Reply:
yes, agree that this is a nice to have that can always be added later when needed